background.ts

```diff
@@ -906,3 +919,28 @@ class BackgroundCommandWorker implements CommandWorkerInterface {
     Electron.app.quit();
   }
 }
+
+/**
+ * Checks if Rancher Desktop was run as root on macOS/Linux,
+ * or as administrator on Windows.
+ */
+function isRoot(): boolean {
+  const platform = os.platform();
+  let isRoot = false;
+
+  if (platform === 'linux' || platform === 'darwin') {
+    if (os.userInfo().uid === 0) {
+      isRoot = true;
+    }
+  } else if (platform === 'win32') {
+    // On windows, running `net session` will throw an error if the process
+    // is not run as administrator. See the following link for more info:
+    // https://stackoverflow.com/questions/4051883/batch-script-how-to-check-for-admin-rights#11995662
+    try {
+      execFileSync('net', ['session'], { stdio: 'ignore' });
+      isRoot = true;
+    } catch { }
```

Comment:
I can't say that I really enjoy using try/catch for control flow. I'm wondering if there's better pattern we could employ to get a return value from `execFileSync` to set `isRoot` 🤔